app/controllers/api/v1/web_hooks_controller.rb

```diff
@@ -30,14 +30,24 @@ def remove
   end
 
   def fire
-    webhook = current_user.web_hooks.new(url: @url)
-    @rubygem = Rubygem.find_by_name("gemcutter") unless @rubygem
+    rubygem = @rubygem ? @rubygem : Rubygem.find_by_name("gemcutter")
+    webhook = current_user.web_hooks.find_by_rubygem_id_and_url(rubygem.try(:id), @url)
+    if webhook
+      if webhook.disabled?
+        webhook.re_enable
+        message = webhook.re_enable_message
+      end
+    else
+      webhook = current_user.web_hooks.new(url: @url)
+    end
 
-    if webhook.fire(request.protocol.delete("://"), request.host_with_port, @rubygem,
-      @rubygem.versions.most_recent, false)
-      render text: webhook.deployed_message(@rubygem)
+    if webhook.fire(request.protocol.delete("://"), request.host_with_port, rubygem,
+      rubygem.versions.most_recent, false)
+      message = message ? message + webhook.deployed_message(rubygem) : webhook.deployed_message(rubygem)
```

Comment:
@simi we are making changes to [`webhook#fire`](http://guides.rubygems.org/rubygems-org-api/#post---apiv1webhooksfire) and the endpoint is only meant for testing purposes.